File: client/core/src/com/mygdx/bomberman/game/GameState.java
package com.mygdx.bomberman.game;

import com.mygdx.bomberman.entities.mapobject.Bomb;
import com.mygdx.bomberman.entities.mapobject.BombExplosion;
import com.mygdx.bomberman.entities.mapobject.Brick;
import com.mygdx.bomberman.entities.mapobject.Player;
import com.mygdx.bomberman.entities.mapobject.Wall;
import com.mygdx.bomberman.entities.mapobject.pickup.BombPickup;
import java.util.ArrayList;
import java.util.List;

public class GameState {
  private final List<Player> players;
  private final List<Bomb> bombs;
  private final List<Wall> walls;
  private final List<BombExplosion> bombExplosions;
  private final List<Brick> bricks;
  private final List<BombPickup> bombPickups;

  public GameState() {
    players = new ArrayList<>();
    bombs = new ArrayList<>();
    walls = new ArrayList<>();
    bombExplosions = new ArrayList<>();
    bricks = new ArrayList<>();
    bombPickups = new ArrayList<>();
  }

  public List<BombPickup> getBombPickups() {
    return bombPickups;
  }

  public List<Player> getPlayers() {
    return players;
  }

  public List<Bomb> getBombs() {
    return bombs;
  }

  public List<Brick> getBricks() {
    return bricks;
  }

  public List<Wall> getWalls() {
    return walls;
  }

  public List<BombExplosion> getBombExplosions() {
    return bombExplosions;
  }
}

File: client/src/game/GameState.ts
import {Pickup, PickupAdapter} from '../entities/mapobject/Pickup';
import {Bomb, BombAdapter} from '../entities/mapobject/Bomb';
import {BombExplosion, BombExplosionAdapter} from '../entities/mapobject/BombExplosion';
import {Brick, BrickAdapter} from '../entities/mapobject/Brick';
import {Player, PlayerAdapter} from '../entities/mapobject/Player';
import {Wall, WallAdapter} from '../entities/mapobject/Wall';
import {Adapter} from "../entities/Adapter";

export class GameState {
    public players: Player[];
    public bombs: Bomb[];
    public walls: Wall[];
    public bombExplosions: BombExplosion[];
    public bricks: Brick[];
    public pickups: Pickup[];

    constructor(players: Player[], bombs: Bomb[], walls: Wall[], bombExplosions: BombExplosion[], bricks: Brick[], pickups: Pickup[]) {
        this.players = players;
        this.bombs = bombs;
        this.walls = walls;
        this.bombExplosions = bombExplosions;
        this.bricks = bricks;
        this.pickups = pickups;
    }
}

export class GameStateAdapter implements Adapter<GameState> {
    adapt(item): GameState {
        if (!item) {
            return null;
        }

        const playerAdapter = new PlayerAdapter()
        const bombAdapter = new BombAdapter()
        const wallAdapter = new WallAdapter()
        const bombExplosionAdapter = new BombExplosionAdapter()
        const brickAdapter = new BrickAdapter()
        const pickupAdapter = new PickupAdapter()

        let players = item.players.map(playerAdapter.adapt)
        let bombs = item.bombs.map(bombAdapter.adapt)
        let walls = item.walls.map(wallAdapter.adapt)
        let bombExplosions = item.bombExplosions.map(bombExplosionAdapter.adapt)
        let bricks = item.bricks.map(brickAdapter.adapt)
        let pickups = item.pickups.map(pickupAdapter.adapt)

        return new GameState(players, bombs, walls, bombExplosions, bricks, pickups);
    }
}
